modules/lib/src/main/resources/assets/js/app/ContentVersions.ts

```diff
@@ -33,6 +48,20 @@ export class ContentVersions {
 
         const activeVersionId: string = contentVersionForViewJson.activeVersion?.contentVersion?.id;
 
-        return new ContentVersions(contentVersions, activeVersionId);
+        let publishedTime: number;
+        let publishedVersionId: string;
+        contentVersions.forEach(cv => {
+            const pi = cv.getPublishInfo();
+            if (!pi) {
+                return;
+            }
+            let pt = pi.getTimestamp().getTime();
+            if (!publishedTime || pt > publishedTime) {
+                publishedTime = pt;
+                publishedVersionId = cv.getId();
+            }
+        });
+
+        return new ContentVersions(contentVersions, activeVersionId, publishedVersionId);
```

Comment:
This is not the right place to do it. This should be done in the constructor of `ContentVersions`, not here. Also all "pi", "pt" and so on must be replaced with human-readable names.

This should also be done differently:
```
const publishedVersions = contentVersions
    .filter(version => version.isPublished())
    .sort((version1, version2) => version2.getTimestamp().getTime() - version1.getTimestamp().getTime());
this.publishedVersionId = publishedVersions[0]?.getId();
```

Reply:
Agree with the pt, pi, and the placement, but don't see a point in increasing execution time:
your example does 2 cycles (one of them being sorting that we don't need), while I did it in 1 go.